WordPress/gutenberg
# Paragraph block text capitalize

## What?
Fixes https://github.com/WordPress/gutenberg/issues/63261

The issue was that the keyboard doesn't start capitalized (uppercase) in the paragraph block on iOS web.

## Why?
The paragraph block did not show the keyboard in uppercase mode by default, causing an inconsistency with expected behavior when typing.

## How?
The fix I’ve implemented ensures that the first letter of the paragraph text is capitalized when the block is focused. This is achieved by modifying the content of the paragraph block when the user starts typing. However, instead of making the entire keyboard input uppercase, I have implemented a solution where the first letter of the content is capitalized. The solution applies for all web and mobile devices and not only for IOS.

I’m not sure whether this is the ideal solution, as the original request was to ensure the keyboard itself starts in uppercase mode. I’d appreciate feedback from reviewers and commenters on whether this approach is acceptable or if a different solution is needed.

## Testing Instructions

- Open a post or page in the WordPress editor.
- Insert a paragraph block.
- Focus on the paragraph block.
- Start typing in the paragraph block and verify that the first letter of the text is automatically capitalized.
- Confirm that the text behavior is consistent across devices, specifically on iOS web.

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/user-attachments/assets/f75fc394-a8c9-4b53-aef8-8714fe693785
